### PR TITLE
GH-95 fix nil error for xray_violations_report

### PR DESCRIPTION
**Pull request by danielmkn**

### CODE REVIEW

File: pkg/xray/reports.go [hunk 1]
```diff
@@ -525,15 +525,15 @@ func unpackViolationsFilters(filter *schema.Set) *Filters {
 		filters.PolicyNames = util.CastToStringArr(m["policy_names"].(*schema.Set).List())
 		filters.Severities = util.CastToStringArr(m["severities"].(*schema.Set).List())
 
-		if m["updated"] != nil {
+		if m["updated"].(*schema.Set).Len() > 0 {
```
Comment (alexhung) [inline pkg/xray/reports.go:528]: Do we still need to check for nil first? e.g. `if m["updated"] != nil && m["updated"].(*schema.Set).Len() > 0`

**Review by Turhsus — DISMISSED** [review]
LGTM

File: pkg/xray/reports.go (hunk 1 above)
Comment (danielmkn) [reply pkg/xray/reports.go:528]: It's never actually a `nil`, I checked. That was a reason it crashed and user created a ticket.

File: pkg/xray/resource_xray_report_test.go [hunk 2]
```diff
@@ -41,33 +41,110 @@ var opRisksFilterFields = map[string]interface{}{
...
+					"max_score": 9,
+				},
+				"summary_contains": "kernel",
+				"has_remediation":  true,
+			},
+			"license_filters": map[string]interface{}{
+				"unknown":       false,
+				"unrecognized":  true,
+				"license_names": []interface{}{"Apache", "MIT"}, // conflicts with license_patterns
+				//"license_patterns": []interface{}{"*Apache*", "The Apache*"},
```
Comment (alexhung) [inline pkg/xray/resource_xray_report_test.go:71]: Is this still needed?

Comment (danielmkn) [reply pkg/xray/resource_xray_report_test.go:71]: I'd say yes. Let's keep it.

**Review by alexhung — APPROVED** [review]
:shipit:

File: pkg/xray/resource_xray_report_test.go (hunk 2 above)
Comment (danielmkn) [reply pkg/xray/resource_xray_report_test.go]: I've removed all commented-out attributes, and kept all permutations in the test.

**Review by alexhung — APPROVED** [review]

**Pull request merged**